In [11]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Basic Pitch Report

### Flute

Average Results:
Precision: 0.000
Recall: 0.000
F1: 0.000
Onset Precision: 0.206
Onset Recall: 0.163
Onset F1: 0.298
Onset-Offset F1: 0.000

In [12]:
# compute summaries
df = pd.read_csv("evaluation_results_basic_pitch_flute.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision  Recall    F1   Onset F1  Onset-Offset F1
count       58.0    58.0  58.0  58.000000             58.0
mean         0.0     0.0   0.0   0.298166              0.0
std          0.0     0.0   0.0   0.107031              0.0
min          0.0     0.0   0.0   0.000000              0.0
25%          0.0     0.0   0.0   0.236842              0.0
50%          0.0     0.0   0.0   0.306043              0.0
75%          0.0     0.0   0.0   0.333333              0.0
max          0.0     0.0   0.0   0.583333              0.0

### Tenor Saxophone

Average Results:
Precision: 0.148
Recall: 0.314
F1: 0.198
Onset Precision: 0.337
Onset Recall: 0.249
Onset F1: 0.550
Onset-Offset F1: 0.086

In [13]:
# compute summaries
df = pd.read_csv("evaluation_results_basic_pitch_tenor_saxophone.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  47.000000  47.000000  47.000000  47.000000        47.000000
mean    0.147879   0.313514   0.197753   0.549920         0.086262
std     0.138050   0.287669   0.181135   0.129881         0.095523
min     0.000000   0.000000   0.000000   0.222222         0.000000
25%     0.000000   0.000000   0.000000   0.481481         0.000000
50%     0.161905   0.444444   0.239521   0.541667         0.091954
75%     0.264914   0.567982   0.366359   0.666667         0.134234
max     0.358974   0.736842   0.482759   0.750000         0.393939

### Trumpet

Average Results:
Precision: 0.069
Recall: 0.154
F1: 0.095
Onset Precision: 0.295
Onset Recall: 0.220
Onset F1: 0.475
Onset-Offset F1: 0.049

In [14]:
# compute summaries
df = pd.read_csv("evaluation_results_basic_pitch_trumpet.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  58.000000  58.000000  58.000000  58.000000        58.000000
mean    0.069058   0.153887   0.094764   0.475499         0.049462
std     0.114598   0.247267   0.155084   0.157934         0.091491
min     0.000000   0.000000   0.000000   0.037037         0.000000
25%     0.000000   0.000000   0.000000   0.371528         0.000000
50%     0.000000   0.000000   0.000000   0.481481         0.000000
75%     0.170430   0.411184   0.239076   0.590278         0.064352
max     0.391304   0.710526   0.504673   0.791667         0.355140

### Violin

Average Results:
Precision: 0.000
Recall: 0.000
F1: 0.000
Onset Precision: 0.278
Onset Recall: 0.189
Onset F1: 0.568
Onset-Offset F1: 0.000



In [15]:
# compute summaries
df = pd.read_csv("evaluation_results_basic_pitch_violin.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision  Recall    F1   Onset F1  Onset-Offset F1
count       58.0    58.0  58.0  58.000000             58.0
mean         0.0     0.0   0.0   0.567907              0.0
std          0.0     0.0   0.0   0.153777              0.0
min          0.0     0.0   0.0   0.148148              0.0
25%          0.0     0.0   0.0   0.500000              0.0
50%          0.0     0.0   0.0   0.578947              0.0
75%          0.0     0.0   0.0   0.702303              0.0
max          0.0     0.0   0.0   0.833333              0.0

# MT3

### Flute

Average Results:
Precision: 0.002
Recall: 0.008
F1: 0.004
Onset Precision: 0.227
Onset Recall: 0.160
Onset F1: 0.456
Onset-Offset F1: 0.000

In [17]:
# compute summaries
df = pd.read_csv("evaluation_results_MT3_flute.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  58.000000  58.000000  58.000000  58.000000        58.000000
mean    0.002308   0.007864   0.003504   0.455977         0.000477
std     0.005798   0.018726   0.008642   0.156939         0.002611
min     0.000000   0.000000   0.000000   0.052632         0.000000
25%     0.000000   0.000000   0.000000   0.368908         0.000000
50%     0.000000   0.000000   0.000000   0.452851         0.000000
75%     0.000000   0.000000   0.000000   0.549890         0.000000
max     0.025000   0.083333   0.037736   0.777778         0.016949

### Tenor Saxophone

Average Results:
Precision: 0.107
Recall: 0.249
F1: 0.145
Onset Precision: 0.263
Onset Recall: 0.184
Onset F1: 0.542
Onset-Offset F1: 0.074

In [18]:
# compute summaries
df = pd.read_csv("evaluation_results_MT3_tenor_saxophone.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  47.000000  47.000000  47.000000  47.000000        47.000000
mean    0.106834   0.248605   0.144501   0.541973         0.073739
std     0.110630   0.214912   0.138113   0.138697         0.086103
min     0.000000   0.000000   0.000000   0.315789         0.000000
25%     0.007120   0.031676   0.011698   0.432749         0.000000
50%     0.080000   0.289474   0.133858   0.552632         0.047059
75%     0.187769   0.427388   0.251316   0.627315         0.137934
max     0.466667   0.625000   0.491228   0.875000         0.350877

### Trumpet

Average Results:
Precision: 0.087
Recall: 0.174
F1: 0.114
Onset Precision: 0.341
Onset Recall: 0.255
Onset F1: 0.578
Onset-Offset F1: 0.067

In [19]:
# compute summaries
df = pd.read_csv("evaluation_results_MT3_trumpet.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  58.000000  58.000000  58.000000  58.000000        58.000000
mean    0.086893   0.173628   0.113850   0.578040         0.066945
std     0.141225   0.266580   0.180430   0.159446         0.122513
min     0.000000   0.000000   0.000000   0.105263         0.000000
25%     0.000000   0.000000   0.000000   0.486111         0.000000
50%     0.000000   0.000000   0.000000   0.578947         0.000000
75%     0.154866   0.447368   0.236467   0.666667         0.084046
max     0.472222   0.791667   0.575758   0.875000         0.500000

### Violin

Average Results:
Precision: 0.002
Recall: 0.004
F1: 0.002
Onset Precision: 0.278
Onset Recall: 0.191
Onset F1: 0.578
Onset-Offset F1: 0.000

In [20]:
# compute summaries
df = pd.read_csv("evaluation_results_MT3_violin.csv")
summary = df.describe()[["Precision", "Recall", "F1", "Onset F1", "Onset-Offset F1"]]
summary

Precision     Recall         F1   Onset F1  Onset-Offset F1
count  58.000000  58.000000  58.000000  58.000000        58.000000
mean    0.001710   0.004420   0.002432   0.577943         0.000454
std     0.004923   0.012261   0.006914   0.157379         0.003455
min     0.000000   0.000000   0.000000   0.052632         0.000000
25%     0.000000   0.000000   0.000000   0.500000         0.000000
50%     0.000000   0.000000   0.000000   0.587963         0.000000
75%     0.000000   0.000000   0.000000   0.666667         0.000000
max     0.019231   0.041667   0.026316   0.916667         0.026316